pandas/tests/io/json/test_pandas.py

```diff
@@ -82,489 +80,379 @@ def setup(self, datapath):
+    @pytest.mark.parametrize("orient", ["index", "columns", "records"])
+    def test_frame_non_unique_columns_raises(self, orient):
         df = DataFrame([["a", "b"], ["c", "d"]], index=[1, 2], columns=["x", "x"])
 
-        msg = "DataFrame columns must be unique for orient='index'"
-        with pytest.raises(ValueError, match=msg):
-            df.to_json(orient="index")
-        msg = "DataFrame columns must be unique for orient='columns'"
-        with pytest.raises(ValueError, match=msg):
-            df.to_json(orient="columns")
-        msg = "DataFrame columns must be unique for orient='records'"
+        msg = "DataFrame columns must be unique for orient='{}'".format(orient)
         with pytest.raises(ValueError, match=msg):
-            df.to_json(orient="records")
+            df.to_json(orient=orient)
 
-        assert_frame_equal(
-            df, read_json(df.to_json(orient="split"), orient="split", dtype=False)
+    def test_frame_default_orient(self):
+        assert self.frame.to_json() == self.frame.to_json(orient="columns")
+
+    @pytest.mark.parametrize("dtype", [False, float])
+    @pytest.mark.parametrize("convert_axes", [True, False])
+    @pytest.mark.parametrize("numpy", [True, False])
+    def test_roundtrip_simple(self, orient, convert_axes, numpy, dtype):
+        data = self.frame.to_json(orient=orient)
+        result = pd.read_json(
+            data, orient=orient, convert_axes=convert_axes, numpy=numpy, dtype=dtype
         )
-        unser = read_json(df.to_json(orient="values"), orient="values")
-        tm.assert_numpy_array_equal(df.values, unser.values)
 
-        # GH4377; duplicate columns not processing correctly
-        df = DataFrame([["a", "b"], ["c", "d"]], index=[1, 2], columns=["x", "y"])
-        result = read_json(df.to_json(orient="split"), orient="split")
-        assert_frame_equal(result, df)
+        expected = self.frame.copy()
 
-        def _check(df):
-            result = read_json(
-                df.to_json(orient="split"), orient="split", convert_dates=["x"]
-            )
-            assert_frame_equal(result, df)
+        if not numpy and (orient == "index" or (PY35 and orient == "columns")):
+            # TODO: debug why sort is required
+            expected = expected.sort_index()
 
-        for o in [
-            [["a", "b"], ["c", "d"]],
-            [[1.5, 2.5], [3.5, 4.5]],
-            [[1, 2.5], [3, 4.5]],
-            [[Timestamp("20130101"), 3.5], [Timestamp("20130102"), 4.5]],
-        ]:
-            _check(DataFrame(o, index=[1, 2], columns=["x", "x"]))
-
-    def test_frame_from_json_to_json(self):
-        def _check_orient(
-            df,
-            orient,
-            dtype=None,
-            numpy=False,
-            convert_axes=True,
-            check_dtype=True,
-            raise_ok=None,
-            sort=None,
-            check_index_type=True,
-            check_column_type=True,
-            check_numpy_dtype=False,
-        ):
```

Comment:
@WillAyd is there anything worth salvaging/combining from the two _check_orient funcs?  or easier just to implement piecewise in the refactored tests?

Reply:
Not particularly that I've found. The Series one (as you noted) is much simpler, but the frame one looks to be a lot of spaghetti code that loosely gets everything to pass. The parametrization is a lot stricter, but save the edge cases marked with TODO should be way more consistent